From 62398265565568b325a4f865e887755910b1dd7a Mon Sep 17 00:00:00 2001
From: "fernandis, williamx" <williamx.fernandis@intel.com>
Date: Mon, 16 Oct 2023 17:40:33 +0000
Subject: [PATCH] Fix redundancy attribute issue for PowerSupply URI

An additional "Redundancy" attribute gets added to the response
of GET method on PowerSupply URI.
The PowerSupply schema doesn't define "Redundancy" as one of its
attribute and hence causes Redfish Validator Failure.
In the current code, while querying for PowerSupply details,
there is an additional call to getChassisData() which adds the
"Redundancy" attribute. This function call is not needed for
PowerSupply details,this commit removes the getChassisData() function
call which resolves the Redfish Validator Issue.

Tested:
1. Redfish validator is passed with this change.
2. GET: /redfish/v1/Chassis/Slot1_BHS2U8X4U2_HSBP_1/PowerSubsystem/
PowerSupplies/SOLUM_CO_IS162F22_PSU2
Response :
{
"@odata.id": "/redfish/v1/Chassis/Slot1_BHS2U8X4U2_HSBP_1/
PowerSubsystem/PowerSupplies/SOLUM_CO_IS162F22_PSU2",
"@odata.type": "#PowerSupply.v1_5_0.PowerSupply",
"Id": "SOLUM_CO_IS162F22_PSU2",
"Manufacturer": "SOLUM CO., LTD. ",
"Model": "IS162F22",
"Name": "SOLUM CO IS162F22 PSU2",
"PartNumber": "G36234-018",
"SerialNumber": "PSSF162202ACNS1622A4BM9Q0292",
"Status": {
"Health": "OK",
"State": "Enabled"
}
}

Signed-off-by: fernandis, williamx <williamx.fernandis@intel.com>

---
 redfish-core/lib/power_supply.hpp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/redfish-core/lib/power_supply.hpp b/redfish-core/lib/power_supply.hpp
index 0c2f2e57..2032a699 100644
--- a/redfish-core/lib/power_supply.hpp
+++ b/redfish-core/lib/power_supply.hpp
@@ -479,7 +479,9 @@ inline void
             "</redfish/v1/JsonSchemas/PowerSupply/PowerSupply.json>; rel=describedby");
         asyncResp->res.jsonValue["@odata.type"] =
             "#PowerSupply.v1_5_0.PowerSupply";
-        asyncResp->res.jsonValue["Name"] = "Power Supply";
+        std::string powerSupplyName = powerSupplyId;
+        std::replace(powerSupplyName.begin(), powerSupplyName.end(), '_', ' ');
+        asyncResp->res.jsonValue["Name"] = std::move(powerSupplyName);
         asyncResp->res.jsonValue["Id"] = powerSupplyId;
         asyncResp->res.jsonValue["@odata.id"] = boost::urls::format(
             "/redfish/v1/Chassis/{}/PowerSubsystem/PowerSupplies/{}", chassisId,
@@ -487,11 +489,7 @@ inline void
 
         asyncResp->res.jsonValue["Status"]["State"] = "Enabled";
         asyncResp->res.jsonValue["Status"]["Health"] = "OK";
-        auto sensorAsyncResp = std::make_shared<SensorsAsyncResp>(
-            asyncResp, chassisId, sensors::dbus::powerPaths,
-            sensors::node::power);
 
-        getChassisData(sensorAsyncResp);
         dbus::utility::getDbusObject(
             powerSupplyPath, powerSupplyInterface,
             [asyncResp,
-- 
2.34.1

